client/state/data-layer/wpcom/jetpack-onboarding/index.js

```diff
@@ -12,9 +12,63 @@ import { translate } from 'i18n-calypso';
 import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
 import { errorNotice } from 'state/notices/actions';
 import { http } from 'state/data-layer/wpcom-http/actions';
-import { JETPACK_ONBOARDING_SETTINGS_SAVE } from 'state/action-types';
+import {
+	JETPACK_ONBOARDING_SETTINGS_REQUEST,
+	JETPACK_ONBOARDING_SETTINGS_SAVE,
+} from 'state/action-types';
+import { getUnconnectedSite } from 'state/selectors';
 import { updateJetpackOnboardingSettings } from 'state/jetpack-onboarding/actions';
 
+export const fromApi = response => {
+	if ( ! response.data || ! response.data.onboarding ) {
+		throw new Error( 'missing onboarding settings' );
+	}
+
+	return response.data.onboarding;
```

Comment:
now that I see this, we could probably just return the nested property lookup and let JS throw if it doesn't exist

```js
export const fromApi = response => response.data.onboarding;
```

at which point it seems silly to have it in a separate function, even sillier to add unit tests for. what are we going to test? that JavaScript throws on property access from `undefined`?

```js
dispatchRequest( request, save, amounce, { fromApi: r => r.data.onboarding } )
```